pkg/node/node.go

```diff
@@ -366,88 +364,54 @@ func NewBee(o Options) (*Bee, error) {
 		b.debugAPIServer = debugAPIServer
 	}
 
-	overlays, err := addressbook.Overlays()
+	addresses, err := addressbook.Addresses()
 	if err != nil {
 		return nil, fmt.Errorf("addressbook overlays: %w", err)
 	}
 
 	var count int32
 	var wg sync.WaitGroup
-	jobsC := make(chan struct{}, 16)
-	for _, o := range overlays {
-		jobsC <- struct{}{}
-		wg.Add(1)
-		go func(overlay swarm.Address) {
-			defer func() {
-				<-jobsC
-			}()
 
+	// add the peers to topology and allow it to connect independantly
+	for _, o := range addresses {
+		_ = topologyDriver.AddPeer(p2pCtx, o.Overlay)
+	}
+
+	// Connect bootnodes (?? they would probably be persisted with the other nodes to the address book on shutdown anyway)
+	for _, a := range o.Bootnodes {
+		wg.Add(1)
+		go func(a string) {
 			defer wg.Done()
-			if err := topologyDriver.AddPeer(p2pCtx, overlay); err != nil {
-				logger.Debugf("topology add peer fail %s: %v", overlay, err)
-				logger.Warningf("topology add peer %s", overlay)
+			addr, err := ma.NewMultiaddr(a)
+			if err != nil {
+				logger.Debugf("multiaddress fail %s: %v", a, err)
+				logger.Warningf("connect to bootnode %s", a)
 				return
 			}
-
-			atomic.AddInt32(&count, 1)
-		}(o)
-	}
-
-	wg.Wait()
-
-	// Connect bootnodes if no nodes from the addressbook was sucesufully added to topology
-	if count == 0 {
-		for _, a := range o.Bootnodes {
-			wg.Add(1)
-			go func(a string) {
-				defer wg.Done()
-				addr, err := ma.NewMultiaddr(a)
+			if _, err := p2p.Discover(p2pCtx, addr, func(addr ma.Multiaddr) (stop bool, err error) {
+				logger.Tracef("connecting to bootnode %s", addr)
+				_, err = p2ps.Connect(p2pCtx, addr, true)
 				if err != nil {
-					logger.Debugf("multiaddress fail %s: %v", a, err)
-					logger.Warningf("connect to bootnode %s", a)
-					return
-				}
-				var count int
```

Comment:
since this is shadowing the previous `count`, the condition below at line `439` would never be hit, since it would always be 1, since this is defined within the new goroutine